### PR TITLE
Updates FlyWay SQLs according to KeyCloak

### PR DESCRIPTION
**Pull request by KaiVolland**
This updates the flyway sqls according to the changes to the models of `User`, `Role`, etc.

It includes a hack to the SecurityContextUtil.

Currently the `userRepository.findByKeycloakId` returns an user but it should return an `Optional`. We should fix this is a followup PR.

### CODE REVIEW

File: shogun-boot/src/main/resources/db/migration/V1.0.2__Init_admin.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+--);
+--
+--INSERT INTO userclasspermissions VALUES (
+--    nextval('hibernate_sequence'),
+--    NOW()::timestamp,
+--    NOW()::timestamp,
+--    'de.terrestris.shogun.lib.model.Group',
+--    (SELECT id FROM permissions WHERE name = 'ADMIN'),
+--    (SELECT id FROM users WHERE username = 'admin')
+--);
```
Comment (dnlkoch) [inline shogun-boot/src/main/resources/db/migration/V1.0.2__Init_admin.sql]: Is there any reason why we keep this file?

Comment (KaiVolland) [reply shogun-boot/src/main/resources/db/migration/V1.0.2__Init_admin.sql]: Nope. I'll remove it.

File: shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql [hunk 2]
```diff
@@ -53,41 +53,24 @@ CREATE TABLE files (
...
     details jsonb,
-    email text NOT NULL,
-    enabled boolean,
-    password text,
-    username text NOT NULL,
     CONSTRAINT users_pkey PRIMARY KEY (id),
-    CONSTRAINT users_unique_email UNIQUE (email),
-    CONSTRAINT users_unique_id UNIQUE (id),
-    CONSTRAINT users_unique_username UNIQUE (username)
+    CONSTRAINT users_unique_id UNIQUE (id)
```
Comment (hwbllmnn) [inline shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql]: You can simplify the unique and pk constraints by using `id bigserial` and annotating the column with
```
@Column
@GeneratedValue(strategy = GenerationType.IDENTITY)
@Id
```

File: shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql [hunk 3]
```diff
@@ -53,41 +53,24 @@ CREATE TABLE files (
...
 );
 
 CREATE TABLE groups (
     id bigint NOT NULL,
```
Comment (hwbllmnn) [inline shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql]: Same here

File: shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql (hunk 2 above)
Comment (ahennr) [reply shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql]: Herewith, sequential IDs would be generated automatically per table, wouldn't they?

Comment (KaiVolland) [reply shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql]: so i can remove the constraint lines?

Comment (hwbllmnn) [reply shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql]: Yep. `id bigserial` (64 bit int) or `id serial` (32 bit int) will add the constraints, set the field as primary key and add a new sequence. This has the added benefit that objects in all tables start with id `1`.

Comment (KaiVolland) [reply shogun-boot/src/main/resources/db/migration/V1.0.0__Baseline.sql]: I'll change this in an follow up PR as this affects all ids.

**Comment by buehner** [comment]
We will address the hints of @hwbllmnn in a follow up PR

**Pull request merged**